eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,144 @@
+param(
+  [string] $InputJsonPath = $GenerateInput,
+  [string] $OutputJsonPath = $GenerateOutput,
+  [string] $RepoRoot = "${PSScriptRoot}/../.."
+)
+
+$inputFileContent = Get-Content $InputJsonPath | ConvertFrom-Json
+$inputFilePaths = $inputFileContent.changedFiles;
```

Comment:
If `changedFiles` is a single object, `$inputFilePaths` will be a string instead of a collection and the `+=` below will fail. Instead, do this:

```powershell
$inputFilePaths = , $inputFileContent.changedFiles
```

The comma is an easy way to force it to be a collection regardless of the rvalue.

Reply:
```
"changedFiles": [
    "specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json"
  ]
```

`changedFiles` is an array. Even if it will have a single object, `$inputFilePaths` will always be a collection. Please let me know if I'm missing anything.

Reply:
PowerShell has some weird behaviors in many places. I'm not sure about `ConvertFrom-Json` specifically, but a lot of other times a single element gets treated as a scalar. I wouldn't change it. Another option is like this:

```powershell
[string[]] $inputFilePaths = $inputFileContent.changedFiles
```